### ISSUE TITLE
Provide example with ImGui and GLFW

### ISSUE THREAD

**Issue by modoodalvi** [treeform/pixie#361]
Expand [GLFW example ](https://github.com/treeform/pixie/blob/master/examples/realtime_glfw.nim) with ImGui support similar to the example provided by [nimgl](https://github.com/nimgl/nimgl/blob/master/examples/tglfw.nim).

**Comment by guzba** [comment]
If you are using ImGui for rendering I don't think you'd have need for Pixie. Could you explain a bit about how you're using them together?

**Comment by treeform** [comment]
Sorry I don't think we want to write example for every possible library out there. 

This request does not make sense for me, I think ImGui is kind of like competitor to pixie?

Closing this. Feel free to reopen if we are missing some thing?

**Comment by modoodalvi** [comment]
I was actually looking to port my project to pixie to see how it compares to my current Skia setup, but couldn't stitch together a working example. 

I am currently using Skia with GLFW and DearImGui on a project. I found that the combination using a 2D graphics with ImGui pretty common, such that Skia offers a [ImGui layer](https://github.com/google/skia/blob/main/tools/viewer/ImGuiLayer.cpp). In general I saw a lot of projects where it is used complementary. ImGui to configure, edit, visualize and debug  the scene and the graphics library to actually render the scene. Having sponsors like Google, Adobe, Epic, Ubisoft, Nvidia among others [DearImgui](https://github.com/ocornut/imgui/wiki/Sponsors) doesn't seem like a random library out there. 

As with the GLFW example proving pixie to be useful in "realtime" scenarios, I thought having an example with ImGui will prove helpful to push pixies adoption in scenarios where ImGui usage together with a graphics library is simply existential.
